core/src/com/cloud/storage/template/QCOW2Processor.java

```diff
@@ -75,6 +76,16 @@ public FormatInfo process(String templatePath, ImageFormat format, String templa
 
     @Override
     public long getVirtualSize(File file) throws IOException {
+        try {
+            long size = getTemplateVirtualSize(file);
```

Comment:
The utils one should be a recent implementation. I wonder why the author hasnt done cleanup and left two implementation of the same. I will take a look at the code later and see if it can be reused here.